### PR TITLE
SLCORE-498 Properly answer to CORS preflight requests

### PR DESCRIPTION
**Pull request by nquinquenel**

### CODE REVIEW

File: core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java [hunk 1]
```diff
@@ -45,10 +46,14 @@ public void sendInformation(ClassicHttpResponse classicHttpResponse) throws Http
       public void submitResponse(ClassicHttpResponse classicHttpResponse) throws HttpException, IOException {
         if (origin != null) {
           classicHttpResponse.addHeader("Access-Control-Allow-Origin", origin.getValue());
+          if (Method.OPTIONS.name().equalsIgnoreCase(method)) {
+            // Private network resources should not be accessible to all origins
```
[the remark this answers is not in the document]
Comment (nquinquenel) [reply core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java:50]: You are kinda right. I wanted to emphasize that we should use `Access-Control-Allow-Private-Network` only when a specific origin is set (the server used by the current binding). There is no default value for the `Allow-Origin` header so it's fine, but we should avoid using `*` and allow `Private-Network` at the same time

[the remark this answers is not in the document]
Comment (nquinquenel) [reply core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java]: I improved the comment

File: core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java [hunk 2]
```diff
@@ -45,10 +46,14 @@ public void sendInformation(ClassicHttpResponse classicHttpResponse) throws Http
       public void submitResponse(ClassicHttpResponse classicHttpResponse) throws HttpException, IOException {
         if (origin != null) {
           classicHttpResponse.addHeader("Access-Control-Allow-Origin", origin.getValue());
+          if (Method.OPTIONS.name().equalsIgnoreCase(method)) {
+            // Private network resources should only be accessible to the server currently connected to
```
Comment (damien-urruty-sonarsource) [inline core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java:50]: With your change, we set the `Access-Control-Allow-Private-Network` header if the `Origin` header is present in the request, which I think is the case most of the time anyway. I don't see a big difference in moving it here, but I might miss something. So the comment is still a bit misleading IMO

**Review by damien-urruty-sonarsource — APPROVED** [review]
I'm not too sure about moving the `Access-Control-Allow-Private-Network` header addition, but I don't think it makes a big difference so LGTM

File: core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java (hunk 2 above)
Comment (nquinquenel) [reply core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java]: Right, it's likely the same in the end, rollbacked to the first state finally 😄

**Pull request merged**